DataDog/datadog-lambda-java
# Add KinesisHeaderable, Add helper method to build JSON from the tracing context

### What does this PR do?

This PR is a continuation of https://github.com/DataDog/datadog-lambda-java/pull/36. 
In this PR added:
- `KinesisHeaderable` - a similar class to `SQSHeaderable`, but for Kinesis Lambda Events.
- `DDLambda.getTraceContextJSONString` - a method to get JSON of the current tracing info. This is needed when you want to pass the DD context through message queues. In our case, we need to it be able to append the tracing info to SQS and Kinesis messages. Usage example:
```java
public void sendMessageToSQS(DDLambda ddl) {
   SendMessageRequest sendMessageRequest = new SendMessageRequest()
     .withQueueUrl("somewhere")
     .withMessageBody("foobar")
     .withMessageAttributes(Map.of(
       SQSHeaderable.DATADOG_ATTRIBUTE_NAME, new MessageAttributeValue().withStringValue(ddLambda.getTraceContextJSONString())
   ));
   sqs.sendMessage(sendMessageRequest);
}
```

### Motivation

Improving usability of `datadog-lambda-java`

### Testing Guidelines

I made unit tests. Please note that there is an added dependency to `com.amazonaws:aws-java-sdk-kinesis:1.11.980` for tests. This is required to be able to build a Kinesis record in the test to make sure it is properly parsed.

### Additional Notes

Please note, that there is a `compileOnly` dependency of `com.amazonaws:aws-java-sdk-kinesis:1.11.980`. This is required to be able to get Kinesis's record `.getData()`. 
### Types of changes


### Checklist
